cpp/oneapi/dal/backend/primitives/selection/test/perf_select_by_rows_dpc.cpp

```diff
@@ -0,0 +1,109 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include <type_traits>
+#include <tuple>
+
+#include "oneapi/dal/backend/primitives/ndarray.hpp"
+#include "oneapi/dal/backend/primitives/selection/select_by_rows.hpp"
+#include "oneapi/dal/test/engine/common.hpp"
+#include "oneapi/dal/test/engine/fixtures.hpp"
+#include "oneapi/dal/test/engine/dataframe.hpp"
+#include "oneapi/dal/table/row_accessor.hpp"
+
+namespace oneapi::dal::backend::primitives::test {
+
+namespace te = dal::test::engine;
+
+template <typename T>
+struct type2str {
+    static const char* name;
+};
+
+#define INSTANTIATE_TYPE_MAP(T) \
+    template <>                 \
+    const char* type2str<T>::name = #T;
+
+INSTANTIATE_TYPE_MAP(float);
+INSTANTIATE_TYPE_MAP(double)
```

Comment:
@makart19, already added the same functionality to his tests, suggest to generalize it and reuse